### PR TITLE
EXPERIMENTAL: Api, ZIP dags

### PR DESCRIPTION
**Pull request by bolkedebruin**
This is an experimental branch. Just to test out a couple of ideas.
- API: for submitting dag files, loading them from the api for distribution, obtaining connection info for a dag
- Fix for the webserver not reloading the dag folder
- Allow ZIP dag files (can also be used for future dependency management by adding a zip to sys.modules) (superseded by #1352)
- some fixes in refreshing files,
- Let the scheduler download dag files if missing or new
- Some kerberos stuff

### CODE REVIEW

File: airflow/hooks/base_hook.py [hunk 1]
```diff
@@ -12,6 +12,13 @@
 from airflow.models import Connection
 from airflow.exceptions import AirflowException
 
+import airflow.configuration as conf
+
+import json
+import requests
+
+base_url = "http://localhost:8080/api/v1/"
```
Comment (r39132) [inline airflow/hooks/base_hook.py:20]: Shouldn't the port come from conf?

Comment (bolkedebruin) [inline airflow/hooks/base_hook.py:20]: @r39132 sure, this really is experimental, ie. trying out new stuff quick 'n dirty. Nothing you want to merge, but maybe make spinoffs in separate PRs (as with the ZIP functionality)

**Comment by ldct** [comment]
@bolkedebruin should we close this? it seems to me that most points have made it to other PRs
